src/WtsApi32.Desktop/WtsApi32+WtsSafeMemoryGuard.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    /// <content>
+    /// Exported functions from the WtsApi32.dll Windows library
+    /// that are available to Desktop apps only.
+    /// </content>
+    public static partial class WtsApi32
+    {
+        /// <summary>
+        /// Base guard for all memory allocated inside WtsApi32.
+        /// </summary>
+        public partial class WtsSafeMemoryGuard : SafeHandle
```

Comment:
I see, and agree they achieve almost the same but these classes might have some optimization and/or safeguards from the jit/runtime as well as they are Microsoft's own implementation and they could expect that. At least this is generally my motivation for preferring the framework's implementation over one's own implementation. At least if it has no clear higher purpose or clear optimization/enhancement over the one offered by the framework.